src/scipp/io/hdf5.py

```diff
@@ -222,9 +222,36 @@ def read(cls, group):
         return var
 
 
+def _write_mapping(parent, mapping, override=None):
+    if override is None:
+        override = {}
+    for i, name in enumerate(mapping):
+        var_group_name = collection_element_name(name, i)
+        if (g := override.get(var_group_name)) is not None:
+            parent[var_group_name] = g
+        else:
+            g = VariableIO.write(group=parent.create_group(var_group_name),
+                                 var=mapping[name])
+            if g is None:
+                del parent[var_group_name]
+            else:
+                g.attrs['name'] = str(name)
+
+
+def _read_mapping(group, override=None):
+    if override is None:
+        override = {}
+    return {
+        g.attrs['name']: override.get(g.attrs['name'], HDF5IO.read(g))
```

Comment:
This always reads `g` because the arguments to `get` are evaluated before calling `get`. If you want to avoid that, you need something like `override.get(name) if name in override else HDF4IO.read(g)`.

Reply:
Oh nice catch!